gef.py

```diff
@@ -4758,6 +4767,68 @@ def __set_repeat_count(self, argv: List[str], from_tty: bool) -> None:
         return
 
 
+@register
+class ArchCommand(GenericCommand):
+    """Manage the current loaded architecture."""
+
+    _cmdline_ = "arch"
+    _syntax_ = f"{_cmdline_} (list|get|set) ..."
+    _example_ = f"{_cmdline_} set X86"
+
+    def __init__(self) -> None:
+        super().__init__(prefix=True)
+        return
+
+    def do_invoke(self, argv: List[str]) -> None:
+        if not argv:
+            self.usage()
+        return
+
+@register
+class ArchGetCommand(GenericCommand):
+    """Get the current loaded architecture."""
+
+    _cmdline_ = "arch get"
+    _syntax_ = f"{_cmdline_}"
+    _example_ = f"{_cmdline_}"
+
+    def do_invoke(self, args: List[str]) -> None:
+        gef_print(f"{Color.greenify('Arch')}: {gef.arch}")
+        gef_print(f"{Color.greenify('Reason')}: {gef.arch_reason}")
+
+
+@register
+class ArchSetCommand(GenericCommand):
+    """Set the current loaded architecture."""
+
+    _cmdline_ = "arch set"
+    _syntax_ = f"{_cmdline_} <arch>"
+    _example_ = f"{_cmdline_} X86"
+
+    def do_invoke(self, args: List[str]) -> None:
+        reset_architecture(args[0].lower() if args else None)
+
+    def complete(self, text: str, word: str) -> List[str]:
+        return sorted(x for x in __registered_architectures__.keys() if
+                       isinstance(x, str) and x.lower().startswith(text.lower().strip()))
+
+@register
+class ArchListCommand(GenericCommand):
+    """List the available architectures."""
+
+    _cmdline_ = "arch list"
+    _syntax_ = f"{_cmdline_}"
+    _example_ = f"{_cmdline_}"
+
+    def do_invoke(self, args: List[str]) -> None:
+        gef_print(Color.greenify("Available architectures:"))
+        for arch in sorted(set(__registered_architectures__.values()), key=lambda x: x.arch):
+            if arch != GenericArchitecture:
+                gef_print(' ' + Color.yellowify(str(arch())))
+                for alias in filter(lambda x: isinstance(x, str), arch.aliases):
```

Comment:
slight preference for comprehensions here e.g. `for alias in (x for x in arch.aliases if isinstance(x, str))`

Reply:
or maybe easiest would be
```python
for alias in arch.aliases:
    if isinstance(alias, str):
        gef_print(f"  {alias}")
```

One liners are nicer but not when they make things more confusing.